src/vs/workbench/services/preferences/browser/preferencesService.ts

```diff
@@ -217,23 +221,49 @@ export class PreferencesService extends Disposable implements IPreferencesServic
 				"textual" : { "classification": "SystemMetaData", "purpose": "FeatureInsight", "isMeasurement": true }
 			}
 		*/
+		const openDefaultKeybindings = !!this.configurationService.getValue('workbench.settings.openDefaultKeybindings');
 		this.telemetryService.publicLog('openKeybindings', { textual });
 		if (textual) {
 			const emptyContents = '// ' + nls.localize('emptyKeybindingsHeader', "Place your key bindings in this file to overwrite the defaults") + '\n[\n]';
 			const editableKeybindings = URI.file(this.environmentService.appKeybindingsPath);
 
 			// Create as needed and open in editor
+			if (openDefaultKeybindings) {
+				return this.createIfNotExists(editableKeybindings, emptyContents).then(() => {
+					const activeEditorGroup = this.editorGroupService.activeGroup;
+					const sideEditorGroup = this.editorGroupService.addGroup(activeEditorGroup.id, GroupDirection.RIGHT);
+
+					return TPromise.join([
+						this.editorService.openEditor({ resource: this.defaultKeybindingsResource, options: { pinned: true, preserveFocus: true }, label: nls.localize('defaultKeybindings', "Default Keybindings"), description: '' }),
+						this.editorService.openEditor({ resource: editableKeybindings, options: { pinned: true } }, sideEditorGroup.id)
+					]).then(editors => void 0);
+				});
+			}
 			return this.createIfNotExists(editableKeybindings, emptyContents).then(() => {
-				const activeEditorGroup = this.editorGroupService.activeGroup;
-				const sideEditorGroup = this.editorGroupService.addGroup(activeEditorGroup.id, GroupDirection.RIGHT);
-
-				return TPromise.join([
-					this.editorService.openEditor({ resource: this.defaultKeybindingsResource, options: { pinned: true, preserveFocus: true }, label: nls.localize('defaultKeybindings', "Default Keybindings"), description: '' }),
-					this.editorService.openEditor({ resource: editableKeybindings, options: { pinned: true } }, sideEditorGroup.id)
-				]).then(editors => void 0);
+				return this.editorService.openEditor({ resource: editableKeybindings, options: { pinned: true } }).then(editors => void 0);
 			});
 		}
-		return this.editorService.openEditor(this.instantiationService.createInstance(KeybindingsEditorInput), { pinned: true }).then(() => null);
+
+		const keybindingsEditorInput = this.instantiationService.createInstance(KeybindingsEditorInput);
```

Comment:
I do not think `openDefaultKeybindings` setting should interfere in Keybindings editor. This setting should decide to open or not the default keybindings file when you open keybindings as a file instead of editor.